### ISSUE TITLE
Setting variable value directly is not equivalent to setting value using '=='

### ISSUE THREAD

**Issue by pgkirsch** [convexengineering/gpkit#454]
I believe the following two code snippets should work the same, but they don't. The first solves just fine. The second returns unknown.

``` python
from gpkit import Variable, Model, SignomialsEnabled, units
import numpy as np

Qv = Variable('Qv', 'N*m', 'Torsion moment imparted by vertical tail')
Lvmax = Variable('Lvmax', 35000, 'N', 'Tail max lift')
bv = Variable('bv', 7, 'm', 'Vertical tail span')
plamv = Variable('plamv', 2.3, '-', 'Tail taper ratio + 2')
qlamv = Variable('qlamv', 1.3, '-', 'Tail taper ratio + 1')
taucone = Variable('taucone', 15000/0.000145,'N/m^2', 'Shear stress in cone')
dRfuse = Variable('dRfuse', 15*0.0254, 'm', 'Offset of lower fuselage cylinder')
Afuse = Variable('Afuse', 'm^2', 'Cross sectional area')
tcone = Variable('tcone', 'm', 'Cone thickness')
Rfuse = Variable('Rfuse', 77*0.0254, 'm', 'Fuselage radius')
plamcone = Variable('plamcone', 1.3, '-', 'Cone taper ratio + 1')
lcone = Variable('lcone', 15*0.3048, 'm', 'Cone length') # added this 
Wcone = Variable('Wcone', 'N', 'Weight of cone')
rhocone = Variable('rhocone', 2700, 'kg/m^3', 'Density of cone material') # [TAS]
g = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')

objective = Wcone
constraints = [
                Afuse >= np.pi*Rfuse**2 + 2*Rfuse*dRfuse,
                Qv == 2*Afuse*taucone*tcone,
                Wcone >= rhocone*g*2*Lvmax*bv*plamv/(3*qlamv)*lcone/(taucone*Rfuse*plamcone),
              ]
m = Model(objective, constraints)
m.solve()
```

The only difference is how the value of Rfuse is specified.

``` python
from gpkit import Variable, Model, SignomialsEnabled, units
import numpy as np

Qv = Variable('Qv', 'N*m', 'Torsion moment imparted by vertical tail')
Lvmax = Variable('Lvmax', 35000, 'N', 'Tail max lift')
bv = Variable('bv', 7, 'm', 'Vertical tail span')
plamv = Variable('plamv', 2.3, '-', 'Tail taper ratio + 2')
qlamv = Variable('qlamv', 1.3, '-', 'Tail taper ratio + 1')
taucone = Variable('taucone', 15000/0.000145,'N/m^2', 'Shear stress in cone')
dRfuse = Variable('dRfuse', 15*0.0254, 'm', 'Offset of lower fuselage cylinder')
Afuse = Variable('Afuse', 'm^2', 'Cross sectional area')
tcone = Variable('tcone', 'm', 'Cone thickness')
Rfuse = Variable('Rfuse', 'm', 'Fuselage radius')
plamcone = Variable('plamcone', 1.3, '-', 'Cone taper ratio + 1')
lcone = Variable('lcone', 15*0.3048, 'm', 'Cone length') # added this 
Wcone = Variable('Wcone', 'N', 'Weight of cone')
rhocone = Variable('rhocone', 2700, 'kg/m^3', 'Density of cone material') # [TAS]
g = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')

objective = Wcone
constraints = [
                Rfuse == 77*0.0254*units.m,
                Afuse >= np.pi*Rfuse**2 + 2*Rfuse*dRfuse,
                Qv == 2*Afuse*taucone*tcone,
                Wcone >= rhocone*g*2*Lvmax*bv*plamv/(3*qlamv)*lcone/(taucone*Rfuse*plamcone),
              ]
m = Model(objective, constraints)
m.solve()
```

**Comment by whoburg** [comment]
Interesting! @bqpd, correct me if I'm wrong, but what gets passed to the solver is different for these two cases. In the first case, the value given for `Rfuse` gets substituted into all nomial expressions by GPkit before the problem is passed to the solver. In the second case, Rfuse is a decision variable chosen by the solver.

That said, it's certainly a surprise for the behavior to be different!

**Comment by bqpd** [comment]
Yup, that's what's happening. I would definitely expect both of these to solve...

**Comment by bqpd** [comment]
CVXOPT says ``Terminated (singular KKT matrix).``

**Comment by bqpd** [comment]
MOSEK solves but than has an error during solution checking.

**Comment by whoburg** [comment]
mosek output:

```RuntimeWarning: Primal solution computed cost did not match solver-returned cost: 133.04766326 vs 0.0```

**Comment by pgkirsch** [comment]
Weird. When I solve with mosek, I get:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/kirschen/blah.py in <module>()
     26               ]
     27 m = Model(objective, constraints)
---> 28 m.solve()

/home/kirschen/anaconda/lib/python2.7/site-packages/gpkit/model.pyc in solve(self, solver, verbosity, skipfailures, *args, **kwargs)
    207         try:
    208             return self._solve("gp", solver, verbosity, skipfailures,
--> 209                                *args, **kwargs)
    210         except ValueError as err:
    211             if err.message == ("GeometricPrograms cannot contain Signomials"):

/home/kirschen/anaconda/lib/python2.7/site-packages/gpkit/model.pyc in _solve(self, programType, solver, verbosity, skipfailures, *args, **kwargs)
    361                                                  verbosity=verbosity)
    362             result = solvefn(*args, **kwargs)
--> 363             solution.append(parse_result(result, constants, beforesubs))
    364         solution.program = self.program
    365         solution.toarray()

/home/kirschen/anaconda/lib/python2.7/site-packages/gpkit/solution_array.pyc in parse_result(result, constants, beforesubs, sweep, linkedsweep, freevar_sensitivity_tolerance, localmodel_sensitivity_requirement)
    242     localcs = (variables[var]**-S for (var, S) in localexp.items())
    243     localc = functools_reduce(mul, localcs, cost)
--> 244     localmodel = Monomial(localexp, localc)
    245 
    246     # vectorvar substitution

/home/kirschen/anaconda/lib/python2.7/site-packages/gpkit/nomials.pyc in __init__(self, exps, cs, require_positive, simplify, **descr)
    109             from . import SIGNOMIALS_ENABLED
    110             if require_positive and not SIGNOMIALS_ENABLED:
--> 111                 raise ValueError("each c must be positive.")
    112         else:
    113             self.__class__ = Posynomial

ValueError: each c must be positive.
```

**Comment by bqpd** [comment]
That's probably the 0.0

**Comment by pgkirsch** [comment]
Right, but I have seen that solver mismatch error before. It doesn't always/usually go to the c must be positive error.

**Comment by pgkirsch** [comment]
Another example of I think the same underlying issue. 3 pieces of code below:
1. Variable value declared (works)
2. Variable value not declared (doesn't work)
3. Same as (2) but I annexed the second half of one of my posynomials (works)

```python
from gpkit import Variable, Model, SignomialsEnabled, units
import numpy as np

# Variables
Afuse =     Variable('Afuse', 'm^2', 'Cross sectional area')
Askin =     Variable('Askin', 'm^2', 'Cross sectional area of skin')
dRfuse =    Variable('dRfuse',15*0.0254, 'm', 'Offset of lower fuselage cylinder')
lnose =     Variable('lnose', 'm', 'Nose length')
lshell =    Variable('lshell', (102-17)*0.3048, 'm', 'Shell length')
#Rfuse =     Variable('Rfuse', 'm', 'Fuselage radius')
Rfuse =     Variable('Rfuse', 77*0.0254, 'm', 'Fuselage radius')
Sbulk =     Variable('Sbulk', 'm^2', 'Bulkhead surface area')
Sfloor =    Variable('Sfloor', 'N', 'Maximum shear in floor beams')
sigth =     Variable('sigth', 'N/m^2', 'Hoop stress in skin')
sigx =      Variable('sigx', 'N/m^2', 'Axial stress in skin')
Snose =     Variable('Snose', 'm^2', 'Nose surface area')
tshell =    Variable('tshell', 'm', 'Shell thickness')
tskin =     Variable('tskin', 'm', 'Skin thickness')
Vbulk =     Variable('Vbulk', 'm^3', 'Bulkhead skin volume')
Vcone =     Variable('Vcone', 'm^3', 'Volume of cone skin')
Vcyl =      Variable('Vcyl', 'm^3', 'Cylinder skin volume')
Vnose =     Variable('Vnose', 'm^3', 'Nose skin volume')
Wshell =    Variable('Wshell', 'N', 'Shell weight')
Wskin =     Variable('Wskin', 'N', 'Skin weight')
xshell1 =   Variable('xshell1', 'm', 'Start of cylinder section') # value subbed as constraint

# Constants ([TAS] means sourced from TASOPT)

dp = Variable('\\Deltap', 76000-24000, 'Pa', 'Pressure difference across fuselage skin')
ffadd = Variable('ffadd', 0.20, '-', 'Skin proportional added weight of local reinforcmenets') # [TAS]
fframe = Variable('fframe', 0.25, '-', 'Fractional weight of frame')
fstring = Variable('fstring', 0.35, '-', 'Fraction of stringers?') # [TAS]
g = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
rE = Variable('rE', 1.0, '-', 'Ratio of stringer/skin moduli') # [TAS]
rhobend = Variable('rhobend', 2700, 'kg/m^3', 'Stringer density') # [TAS]
rhoskin = Variable('rhoskin', 2700, 'kg/m^3', 'Skin density') # [TAS]
sigskin = Variable('sigskin', 15000/0.000145, 'N/m^2', 'Max allowable skin stress') # [TAS]

objective = Wshell

constraints = [
                # Geometry relations
                lnose == 5.182*units.m,

                # Cross section relations
                Rfuse >= 77*0.0254*units.m, # TODO: this needs to be driven by something e.g. number of passengers
                dRfuse >= 15*0.0254*units.m, # TODO: this needs to be driven by something e.g. cargo capacity
                Askin >= 2*np.pi*Rfuse*tskin + 2*dRfuse*tskin,
                #Afuse >= np.pi*Rfuse**2,#+ 2*Rfuse*dRfuse,
                Afuse >= np.pi*Rfuse**2+ 2*Rfuse*dRfuse,
                tshell >= tskin*(1 + rE*fstring*rhoskin/rhobend),

                # Pressure shell loads
                sigx == dp/2*Rfuse/tshell,
                sigth == dp*Rfuse/tskin,
                tskin == dp*Rfuse/sigskin,
                Snose**(8./5) >= (2*np.pi*Rfuse**2)**(8./5) * (1./3 + (2./3)*(lnose/Rfuse)**(8./5)),
                Sbulk == 2*np.pi*Rfuse**2,
                Vcyl == Askin*lshell,
                Vnose == Snose*tskin,
                Vbulk == Sbulk*tskin,
                Wskin >= rhoskin*g*(Vcyl + Vnose + Vbulk),
                Wshell >= Wskin*(1 + fstring + fframe + ffadd),
                ]

m = Model(objective, constraints)
sol = m.solve()
```

```python
from gpkit import Variable, Model, SignomialsEnabled, units
import numpy as np

# Variables
Afuse =     Variable('Afuse', 'm^2', 'Cross sectional area')
Askin =     Variable('Askin', 'm^2', 'Cross sectional area of skin')
#dRfuse =    Variable('dRfuse',15*0.0254, 'm', 'Offset of lower fuselage cylinder')
dRfuse =    Variable('dRfuse', 'm', 'Offset of lower fuselage cylinder')
lnose =     Variable('lnose', 'm', 'Nose length')
lshell =    Variable('lshell', (102-17)*0.3048, 'm', 'Shell length')
Rfuse =     Variable('Rfuse', 'm', 'Fuselage radius')
#Rfuse =     Variable('Rfuse', 77*0.0254, 'm', 'Fuselage radius')
Sbulk =     Variable('Sbulk', 'm^2', 'Bulkhead surface area')
Sfloor =    Variable('Sfloor', 'N', 'Maximum shear in floor beams')
sigth =     Variable('sigth', 'N/m^2', 'Hoop stress in skin')
sigx =      Variable('sigx', 'N/m^2', 'Axial stress in skin')
Snose =     Variable('Snose', 'm^2', 'Nose surface area')
tshell =    Variable('tshell', 'm', 'Shell thickness')
tskin =     Variable('tskin', 'm', 'Skin thickness')
Vbulk =     Variable('Vbulk', 'm^3', 'Bulkhead skin volume')
Vcone =     Variable('Vcone', 'm^3', 'Volume of cone skin')
Vcyl =      Variable('Vcyl', 'm^3', 'Cylinder skin volume')
Vnose =     Variable('Vnose', 'm^3', 'Nose skin volume')
Wshell =    Variable('Wshell', 'N', 'Shell weight')
Wskin =     Variable('Wskin', 'N', 'Skin weight')
xshell1 =   Variable('xshell1', 'm', 'Start of cylinder section') # value subbed as constraint

# Constants ([TAS] means sourced from TASOPT)

dp = Variable('\\Deltap', 76000-24000, 'Pa', 'Pressure difference across fuselage skin')
ffadd = Variable('ffadd', 0.20, '-', 'Skin proportional added weight of local reinforcmenets') # [TAS]
fframe = Variable('fframe', 0.25, '-', 'Fractional weight of frame')
fstring = Variable('fstring', 0.35, '-', 'Fraction of stringers?') # [TAS]
g = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
rE = Variable('rE', 1.0, '-', 'Ratio of stringer/skin moduli') # [TAS]
rhobend = Variable('rhobend', 2700, 'kg/m^3', 'Stringer density') # [TAS]
rhoskin = Variable('rhoskin', 2700, 'kg/m^3', 'Skin density') # [TAS]
sigskin = Variable('sigskin', 15000/0.000145, 'N/m^2', 'Max allowable skin stress') # [TAS]

objective = Wshell

constraints = [
                # Geometry relations
                lnose == 5.182*units.m,

                # Cross section relations
                Rfuse >= 77*0.0254*units.m, # TODO: this needs to be driven by something e.g. number of passengers
                dRfuse >= 15*0.0254*units.m, # TODO: this needs to be driven by something e.g. cargo capacity
                Askin >= 2*np.pi*Rfuse*tskin + 2*dRfuse*tskin,
                #Afuse >= np.pi*Rfuse**2,#+ 2*Rfuse*dRfuse,
                Afuse >= np.pi*Rfuse**2+ 2*Rfuse*dRfuse,
                tshell >= tskin*(1 + rE*fstring*rhoskin/rhobend),

                # Pressure shell loads
                sigx == dp/2*Rfuse/tshell,
                sigth == dp*Rfuse/tskin,
                tskin == dp*Rfuse/sigskin,
                Snose**(8./5) >= (2*np.pi*Rfuse**2)**(8./5) * (1./3 + (2./3)*(lnose/Rfuse)**(8./5)),
                Sbulk == 2*np.pi*Rfuse**2,
                Vcyl == Askin*lshell,
                Vnose == Snose*tskin,
                Vbulk == Sbulk*tskin,
                Wskin >= rhoskin*g*(Vcyl + Vnose + Vbulk),
                Wshell >= Wskin*(1 + fstring + fframe + ffadd),
                ]

m = Model(objective, constraints)
sol = m.solve()
```

```python
from gpkit import Variable, Model, SignomialsEnabled, units
import numpy as np

# Variables
Afuse =     Variable('Afuse', 'm^2', 'Cross sectional area')
Askin =     Variable('Askin', 'm^2', 'Cross sectional area of skin')
#dRfuse =    Variable('dRfuse',15*0.0254, 'm', 'Offset of lower fuselage cylinder')
dRfuse =    Variable('dRfuse', 'm', 'Offset of lower fuselage cylinder')
lnose =     Variable('lnose', 'm', 'Nose length')
lshell =    Variable('lshell', (102-17)*0.3048, 'm', 'Shell length')
Rfuse =     Variable('Rfuse', 'm', 'Fuselage radius')
#Rfuse =     Variable('Rfuse', 77*0.0254, 'm', 'Fuselage radius')
Sbulk =     Variable('Sbulk', 'm^2', 'Bulkhead surface area')
Sfloor =    Variable('Sfloor', 'N', 'Maximum shear in floor beams')
sigth =     Variable('sigth', 'N/m^2', 'Hoop stress in skin')
sigx =      Variable('sigx', 'N/m^2', 'Axial stress in skin')
Snose =     Variable('Snose', 'm^2', 'Nose surface area')
tshell =    Variable('tshell', 'm', 'Shell thickness')
tskin =     Variable('tskin', 'm', 'Skin thickness')
Vbulk =     Variable('Vbulk', 'm^3', 'Bulkhead skin volume')
Vcone =     Variable('Vcone', 'm^3', 'Volume of cone skin')
Vcyl =      Variable('Vcyl', 'm^3', 'Cylinder skin volume')
Vnose =     Variable('Vnose', 'm^3', 'Nose skin volume')
Wshell =    Variable('Wshell', 'N', 'Shell weight')
Wskin =     Variable('Wskin', 'N', 'Skin weight')
xshell1 =   Variable('xshell1', 'm', 'Start of cylinder section') # value subbed as constraint

# Constants ([TAS] means sourced from TASOPT)

dp = Variable('\\Deltap', 76000-24000, 'Pa', 'Pressure difference across fuselage skin')
ffadd = Variable('ffadd', 0.20, '-', 'Skin proportional added weight of local reinforcmenets') # [TAS]
fframe = Variable('fframe', 0.25, '-', 'Fractional weight of frame')
fstring = Variable('fstring', 0.35, '-', 'Fraction of stringers?') # [TAS]
g = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
rE = Variable('rE', 1.0, '-', 'Ratio of stringer/skin moduli') # [TAS]
rhobend = Variable('rhobend', 2700, 'kg/m^3', 'Stringer density') # [TAS]
rhoskin = Variable('rhoskin', 2700, 'kg/m^3', 'Skin density') # [TAS]
sigskin = Variable('sigskin', 15000/0.000145, 'N/m^2', 'Max allowable skin stress') # [TAS]

objective = Wshell

constraints = [
                # Geometry relations
                lnose == 5.182*units.m,

                # Cross section relations
                Rfuse >= 77*0.0254*units.m, # TODO: this needs to be driven by something e.g. number of passengers
                dRfuse >= 15*0.0254*units.m, # TODO: this needs to be driven by something e.g. cargo capacity
                Askin >= 2*np.pi*Rfuse*tskin + 2*dRfuse*tskin,
                Afuse >= np.pi*Rfuse**2,#+ 2*Rfuse*dRfuse,
                #Afuse >= np.pi*Rfuse**2+ 2*Rfuse*dRfuse,
                tshell >= tskin*(1 + rE*fstring*rhoskin/rhobend),

                # Pressure shell loads
                sigx == dp/2*Rfuse/tshell,
                sigth == dp*Rfuse/tskin,
                tskin == dp*Rfuse/sigskin,
                Snose**(8./5) >= (2*np.pi*Rfuse**2)**(8./5) * (1./3 + (2./3)*(lnose/Rfuse)**(8./5)),
                Sbulk == 2*np.pi*Rfuse**2,
                Vcyl == Askin*lshell,
                Vnose == Snose*tskin,
                Vbulk == Sbulk*tskin,
                Wskin >= rhoskin*g*(Vcyl + Vnose + Vbulk),
                Wshell >= Wskin*(1 + fstring + fframe + ffadd),
                ]

m = Model(objective, constraints)
sol = m.solve()
```

**Comment by bqpd** [comment]
This is definitely a solver goof, perhaps related to the singular KKT matrix? MOSEK is doing that thing it does where it returns a cost and sensitivities of 0 (I think there's another issue where that happens), and the whole thing solves on both solvers if you remove the two middle constraints that are irrelevant to the objective. It also solves on MOSEK if you add Afuse to the objective.

**Comment by whoburg** [comment]
Hmm. Was the other time we saw the 0 cost the problem where the feasible set was a single point? My memory is foggy on this.

**Comment by whoburg** [comment]
The issue I'm referring to was #403, though it doesn't seem that was the zero cost one.

**Comment by bqpd** [comment]
Okay, MWE:

```python
from gpkit import Variable, Model

a = Variable('a')
b = Variable('b')
c = Variable('c')
d = Variable('d')
e = Variable('e')

m = Model(a, [c == 1,         # constraint 1
              a >= 1/c,       # constraint 2
              d >= c + c**2,  # constraint 3
              e == d*b])      # constraint 4

m.solve("cvxopt", verbosity=3)
m.solve("mosek", verbosity=3)
```

Ways to get this to work:
  * (cvxopt only) remove constraint 4, mosek returns zeros
  * (mosek only) remove 3, cvxopt has rank issue
  * (mosek only) change the right hand side of 3 to either `c` or `c**2`
  * (both solve) remove both constraints

**Comment by whoburg** [comment]
wow, great MWE. The mosek failures actually seem more troubling -- we should raise this with them. I'm going to see if an even smaller MWE is possible for the mosek failure.

**Comment by whoburg** [comment]
Will raise with mosek once #460 is done

**Comment by bqpd** [comment]
I printed out the .eo file in #460, so no need to wait on that. :p

**Comment by whoburg** [comment]
Thanks. Here it is annotated for this case. Note the comment in the first line -- is our num constraints incorrect?

```python
7  * num constraints, I think this should be 6
5  * vars: 0=a,1=c,2=e,3=b,4=d
8

*c
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00

*p_idxs
0
1
2
3
4
4
5
6

*t j A_tj
0 0 1.00000000000000000000e+00    * objective is a
3 0 -1.00000000000000000000e+00   * a appearing in a>=1/c
1 1 1.00000000000000000000e+00    * c <= 1
2 1 -1.00000000000000000000e+00   * c >= 1
3 1 -1.00000000000000000000e+00   * c appearing in a>=1/c
4 1 2.00000000000000000000e+00    * c appearing in d>=c+c**2 
5 1 1.00000000000000000000e+00    * c appearing in d>=c+c**2
6 2 -1.00000000000000000000e+00   * e appearing in e>=d*b
7 2 1.00000000000000000000e+00    * e appearing in e<=d*b
6 3 1.00000000000000000000e+00    * b appearing in e>=d*b
7 3 -1.00000000000000000000e+00   * b appearing in e<=d*b
4 4 -1.00000000000000000000e+00   * d appearing in d>=c+c**2
5 4 -1.00000000000000000000e+00   * d appearing in d>=c+c**2
6 4 -1.00000000000000000000e+00   * d appearing in e>=d*b
7 4 1.00000000000000000000e+00    * d appearing in e<=d*b
```

**Comment by whoburg** [comment]
And here's the more minimal example (still fails mosek) with constraint 4 removed:

```python
4
3  * vars: 0=a,1=c,2=d
6

*c
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00
1.00000000000000000000e+00

*p_idxs
0
1
2
3
4
4

*t j A_tj
0 0 1.00000000000000000000e+00    * objective is a
3 0 -1.00000000000000000000e+00   * a appearing in a>=1/c
1 1 1.00000000000000000000e+00    * c <= 1
2 1 -1.00000000000000000000e+00   * c >= 1
3 1 -1.00000000000000000000e+00   * c appearing in a>=1/c
4 1 2.00000000000000000000e+00    * c appearing in d>=c+c**2 
5 1 1.00000000000000000000e+00    * c appearing in d>=c+c**2
4 2 -1.00000000000000000000e+00   * d appearing in d>=c+c**2
5 2 -1.00000000000000000000e+00   * d appearing in d>=c+c**2

```

**Comment by whoburg** [comment]
@bqpd, can you confirm the above is what you get when removing constraint 4? I can't find my tmp dir...

**Comment by bqpd** [comment]
You can also pass "path='.'" to not use tmp dirs

**Comment by bqpd** [comment]
Yup, that's right!

**Comment by bqpd** [comment]
But with 5 for num_constraints

**Comment by bqpd** [comment]
Which I think is right, right? 1+ max(p_idxs)

**Comment by whoburg** [comment]
If you look here: http://docs.mosek.com/7.1/capi/Exponential_optimization.html, `expopt1.eo` has `1` for num constraints -- i.e. they don't add 1, i.e. I think our output files are off by one on that first line.

**Comment by bqpd** [comment]
Ah, good call. Easy to fix.
